packages/contracts/src/core/dao/DAO.sol

```diff
@@ -122,6 +154,28 @@ contract DAO is
         __PermissionManager_init(_initialOwner);
     }
 
+    /// @notice Initializes the DAO after an upgrade from a previous version.
+    /// @param _protocolVersion The protocol version number of the previous DAO implementation contract this upgrade is transitioning from.
+    /// @param _initData The initialization data to be passed to via `upgradeToAndCall` (see [ERC-1967](https://docs.openzeppelin.com/contracts/4.x/api/proxy#ERC1967Upgrade)).
+    function initializeFrom(
+        uint8[3] calldata _protocolVersion,
+        bytes calldata _initData
+    ) external reinitializer(2) {
+        _initData; // Silences the unused function parameter warning.
+
+        // Check that the contract is not upgrading from a different major release.
+        if (_protocolVersion[0] != 1) {
+            revert ProtocolVersionUpgradeNotSupported(_protocolVersion);
+        }
+        // Check that the contract is not downgrading from a newer or the same minor release.
+        if (_protocolVersion[1] >= 3) {
+            revert ProtocolVersionUpgradeNotSupported(_protocolVersion);
```

Comment:
```suggestion
        if (_oldVersion[0] != 1) {
            revert ProtocolVersionUpgradeNotSupported(_oldVersion);
        }
        // Check that the contract is not downgrading from a newer or the same minor release.
        if (_oldVersion[1] >= 3) {
            revert ProtocolVersionUpgradeNotSupported(_oldVersion);
```

Reply:
How about the third version number? A DAO might try to downgrade from 1.3.5 to 1.3.0 in the future and this should not be possible